src/transformers/modeling_t5.py

```diff
@@ -731,11 +758,11 @@ def __init__(self, config):
         self.shared = nn.Embedding(config.vocab_size, config.d_model)
 
         encoder_config = copy.deepcopy(config)
-        self.encoder = T5Stack(encoder_config)
+        self.encoder = T5Stack(encoder_config, self.shared)
```

Comment:
This is an important change: To make T5 API similar to Bart, the encoder and decoder should be able to encode input_ids directly. @thomwolf @craffel